packages/coin-tester/src/components/coins/sol/index.tsx

```diff
@@ -0,0 +1,483 @@
+    );
+  };
+
+  const signSplTokenTransaction = () =>
+    handleState(
+      async () => {
+        if (account.length < 1) throw new Error('please get account first');
+
+        const token = new PublicKey(splTokenTransaction.token);
+
+        const fromPubkey = new PublicKey(account);
+
+        const toAccount = new PublicKey(splTokenTransaction.to);
+
+        const fromTokenAccount = await getAssociatedTokenAddr(token, fromPubkey);
+        await checkAssociateTokenAccountValid(fromTokenAccount, token, fromPubkey);
+        const toTokenAccount = await getAssociatedTokenAddr(token, toAccount);
+        await checkAssociateTokenAccountValid(toTokenAccount, token, toAccount);
+
+        const recentBlockhash = (await connection.getRecentBlockhash()).blockhash;
+
+        const tx = TransactionCreator.transferSplToken(
+          account,
+          fromTokenAccount.toString(),
+          toTokenAccount.toString(),
+          recentBlockhash,
+          splTokenTransaction.amount,
+          splTokenTransaction.decimals
+        );
+
+        const appId = localStorage.getItem('appId');
+        if (!appId) throw new Error('No Appid stored, please register!');
+
+        const signedTx = await sol.signTransaction({
+          transport: transport as Transport,
+          appPrivateKey,
+          appId,
+          transaction: tx,
+        });
+
+        const recoveredTx = Transaction.from(signedTx);
+
+        const verifySig = recoveredTx.verifySignatures();
+
+        // signature need to be valid
+        if (!verifySig) throw new Error('Fail to verify signature');
+
+        return connection.sendRawTransaction(recoveredTx.serialize());
+      },
+      (result) => setSplTokenTransaction((prev: any) => ({ ...prev, result }))
+    );
+
+  const signCreateAssociateAccount = async () => {
+    handleState(
+      async () => {
+        if (account.length < 1) throw new Error('please get account first');
+
+        const token = new PublicKey(associateAccTx.token);
+        const owner = new PublicKey(associateAccTx.owner);
+        const associateAccount = await getAssociatedTokenAddr(token, owner);
+
+        // check if associateAccount was not created
+        const accountData = await connection.getAccountInfo(new PublicKey(associateAccount));
+
+        if (accountData) {
+          if (accountData.data.length > 0) {
+            const tokenExt = Buffer.from(accountData.data.slice(0, 32));
+            const ownerExt = Buffer.from(accountData.data.slice(32, 64));
+            if (tokenExt.equals(token.toBuffer()) && ownerExt.equals(owner.toBuffer()))
+              throw new Error('User already have associate account');
+            throw new Error('Fail to get Associate account: account owner or token is incorrect');
+          }
+          throw new Error('Fail to get Associate account: account data empty');
+        }
+
+        const tx = TransactionCreator.createTokenAssociateAccount(
+          account,
```

Comment:
I think `account` should be `owner`.
We should make `feePayer` same as `TokenAssociateAccount` owner.

Reply:
I think that was not necessary since `signer` (also known as `feepayer`) might not be the owner of `TokenAssociateAccount` on this transaction. Which mean you can sign a create association account for any one and take a few sol as cost fee for transaction to be execute.

Reply:
Well, that makes sense.